### ISSUE TITLE
Fraud proofs gossiping

### ISSUE THREAD

**Issue by tzdybal** [celestiaorg/rollmint#4]
Fraud proofs can be produced only by full nodes. They will be delivered to all nodes in the network with a gossiping mechanism. This will ensure that fraud proofs are delivered to all nodes as soon as possible, without delay (like waiting on block finalization). If the grace period is shorter than block time, light clients don’t need to keep historical data.

**Comment by tzdybal** [comment]
Fraud proofs gossiping should use the same libp2p message router as [transaction gossiping](https://github.com/lazyledger/optimint/issues/3).

**Comment by liamsi** [comment]
Note that for testnet we only want to support fraud proofs for the evm settlement "hub".

**Comment by tzdybal** [comment]
Closing in favor of #552.
